### PR TITLE
39 create subscription and success modale

### PR DESCRIPTION
**Pull request by martmull**
- add optional `BILLING_FREE_TRIAL_DURATION_IN_DAYS` default to 14 env variable
- update getProductPrices endpoint to return a list and not a dict
- add a ChooseYourPlan modale that display if IS_SELF_BILLING_ENABLED feature flag is set
- call billing/checkout
- redirect to a plan-required/payment-success modale
- Add a loader to CreateWorkspace modale
image
image

image
image

### CODE REVIEW

File: packages/twenty-front/src/modules/ui/theme/constants/BoxShadowDark.ts [hunk 1]
```diff
@@ -15,4 +15,5 @@ export const BOX_SHADOW_DARK = {
     GRAY_SCALE.gray100,
     0.12,
   )}, 0px 2px 4px 0px ${RGBA(GRAY_SCALE.gray100, 0.04)}`,
+  dark: `-4px 4px 0 -2px ${RGBA(GRAY_SCALE.gray15, 1)}`,
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/ui/theme/constants/BoxShadowDark.ts:18]: is this from Figma UI theme? has this been discussed with @Bonapara ?

**Review by charlesBochet — COMMENTED** [review]
Thanks @martmull

**Comment by Bonapara** [comment]
Also @martmull 

Can you:
- Set the `14 days` in hard
- Edit the values of what's included:

image

File: packages/twenty-front/src/pages/auth/ChooseYourPlan.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      (price) => price.recurring?.interval === 'month',
+    )?.[0];
+    if (monthPrice && monthPrice.unit_amount && price.unit_amount) {
+      return `Save $${(12 * monthPrice.unit_amount - price.unit_amount) / 100}`;
+    }
+    return 'Cancel anytime';
+  };
+
+  const handleButtonClick = () => {
+    fetch(REACT_APP_SERVER_BASE_URL + '/billing/checkout', {
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/auth/ChooseYourPlan.tsx:80]: @martmull why don't we use apollo here? why do we do a fetch?

**Review by charlesBochet — APPROVED** [review]
Bravo!

**Pull request merged**